sale_quotation_sourcing_stock_route_transit/model/purchase.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+#
+#    Author: Yannick Vaucher
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+from openerp import models, api, exceptions, _
+
+
+class Purchase(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.multi
+    def has_unconfirmed_sales(self):
+        Sale = self.env['sale.order']
+        po_lines = self.mapped('order_line')
+
+        # accepted state is added to be compatible with
+        # OCA/vertical-ngo/logistic_order
+        unconfirmed_sales_count = Sale.search_count(
+            [('order_line.sourced_by', 'in', po_lines.ids),
+             ('state', 'in', ('draft', 'sent', 'accepted'))])
+        return bool(unconfirmed_sales_count)
+
+    @api.multi
+    def wkf_confirm_order(self):
+        """ Forbid confirmation of purchase order when there
+        are unconfirmed sale orders sourced by its purchase lines
+        """
+        if self.has_unconfirmed_sales():
+            raise exceptions.Warning(
+                _("You cannot confirm a purchase order. You have to confirm"
```

Comment:
s/sale orders that is/sales orders that are/
s/sale order(s)/sales order(s)/

And better rename field to `unconfirmed_sales` and it can be a many2many, because this can be easily tested also, and you have the complete model for anything you need.